lib/Conversion/LWEToPolynomial/LWEToPolynomial.cpp

```diff
@@ -116,6 +124,177 @@ struct ConvertRLWEDecrypt : public OpConversionPattern<RLWEDecryptOp> {
   }
 };
 
+struct ConvertRLWEEncrypt : public OpConversionPattern<RLWEEncryptOp> {
+  ConvertRLWEEncrypt(mlir::MLIRContext *context)
+      : OpConversionPattern<RLWEEncryptOp>(context) {}
+
+  using OpConversionPattern::OpConversionPattern;
+
+  LogicalResult matchAndRewrite(
+      RLWEEncryptOp op, OpAdaptor adaptor,
+      ConversionPatternRewriter &rewriter) const override {
+    auto loc = op.getLoc();
+    auto input = adaptor.getInput();
+    auto key = adaptor.getKey();
+
+    // TODO: Migrate to new LWE types and plaintext modulus.
+    auto inputT = cast<lwe::RLWEPlaintextType>(op.getInput().getType());
+    auto inputEncoding = inputT.getEncoding();
+    auto cleartextBitwidth =
+        llvm::TypeSwitch<Attribute, int>(inputEncoding)
+            .Case<lwe::BitFieldEncodingAttr,
+                  lwe::UnspecifiedBitFieldEncodingAttr,
+                  lwe::PolynomialEvaluationEncodingAttr,
+                  lwe::PolynomialCoefficientEncodingAttr>(
+                [](auto attr) -> int { return attr.getCleartextBitwidth(); })
+            .Default([](Attribute attr) -> int {
+              llvm_unreachable(
+                  "Unsupported encoding attribute for cleartext bitwidth");
+              return -1;
+            });
+    auto cleartextStart =
+        llvm::TypeSwitch<Attribute, int>(inputEncoding)
+            .Case<lwe::BitFieldEncodingAttr,
+                  lwe::PolynomialEvaluationEncodingAttr,
+                  lwe::PolynomialCoefficientEncodingAttr>(
+                [](auto attr) -> int { return attr.getCleartextStart(); })
+            .Case<lwe::UnspecifiedBitFieldEncodingAttr>([](auto attr) -> int {
+              llvm_unreachable(
+                  "Upsecified Bit Field Encoding Attribute for cleartext "
+                  "start");
+              return -1;
+            })
+            .Default([](Attribute attr) -> int {
+              llvm_unreachable(
+                  "Unsupported encoding attribute for cleartext start");
+              return -1;
+            });
+
+    // Should return failure if cleartextBitwidth or cleartextStart fail.
+    if (cleartextBitwidth == -1 || cleartextStart == -1) {
```

Comment:
You might want to use the `FailureOr<int>` return value for the TypeSwitch, I usually find it cleaner than -1 but it's up to you